plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -262,28 +275,54 @@ func GetPointID(m telegraf.Metric) string {
 }
 
 func (a *Elasticsearch) Write(metrics []telegraf.Metric) error {
+	var ecs_agent, ecs_host, ecs_event map[string]interface{}
 	if len(metrics) == 0 {
 		return nil
 	}
 
 	bulkRequest := a.Client.Bulk()
 
+	if a.ElasticCommonSchema {
+		ecs_host = getEcsHostInfo()
+		ecs_agent = getEcsAgentInfo()
+	}
+
 	for _, metric := range metrics {
 		var name = metric.Name()
 
+		if a.ElasticCommonSchema {
+			ecs_event = getEcsEventInfo(name)
+		}
+
 		// index name has to be re-evaluated each time for telegraf
 		// to send the metric to the correct time-based index
-		indexName := a.GetIndexName(a.IndexName, metric.Time(), a.TagKeys, metric.Tags())
+		indexName := a.GetIndexName(a.IndexName, metric.Time(), a.TagKeys, metric.Tags(), metric.Name())
 
 		m := make(map[string]interface{})
 
 		m["@timestamp"] = metric.Time()
-		m["measurement_name"] = name
-		m["tag"] = metric.Tags()
-		m[name] = metric.Fields()
+		if !a.ElasticCommonSchema {
+			m["measurement_name"] = name
+			m["tag"] = metric.Tags()
+			m[name] = metric.Fields()
+		} else {
+			t := make(map[string]interface{})
+			t["tags"] = metric.Tags()
+			t["metrics"] = map[string]interface{}{name: metric.Fields()}
+			m["telegraf"] = t
+			m["metricset"] = map[string]interface{}{"name": name}
+			m["event"] = ecs_event
+			m["host"] = ecs_host
+			m["agent"] = ecs_agent
+			m["ecs"] = map[string]interface{}{"version": "1.7.0"}
```

Comment:
Is there a way to add a unit test for this new common schema format? How much would we need to mock of the elasticsearch client api?